tools/walletextension/api/routes.go

```diff
@@ -216,3 +237,249 @@ func submitViewingKeyRequestHandler(walletExt *walletextension.WalletExtension,
+		return
+	}
+
+	userID, err := getQueryParameter(userConn.ReadRequestParams(), "u")
+	if err != nil {
+		userConn.HandleError("user ('u') not found in query parameters")
+		walletExt.Logger().Error(fmt.Errorf("user not found in the query params: %w", err).Error())
+		return
+	}
+	userIDBytes, err := hex.DecodeString(userID[2:]) // remove 0x prefix from userID
+	if err != nil {
+		userConn.HandleError("Internal error")
+		walletExt.Logger().Error(fmt.Errorf("error decoding string (%s), %w", userID[2:], err).Error())
+		return
+	}
+
+	address, err := getQueryParameter(userConn.ReadRequestParams(), "a")
+	if err != nil {
+		userConn.HandleError("address ('a') not found in query parameters")
+		walletExt.Logger().Error(fmt.Errorf("address not found in the query params: %w", err).Error())
+		return
+	}
+	addressBytes, err := hex.DecodeString(address[2:]) // remove 0x prefix from address
+	if err != nil {
+		userConn.HandleError("Internal error")
+		walletExt.Logger().Error(fmt.Errorf("error decoding string (%s), %w", address[2:], err).Error())
+		return
+	}
+
+	// todo - this can be optimised and done in the database if we will have users with large number of accounts
+	// get all the accounts for the selected user
+	accounts, err := walletExt.Storage.GetAccounts(userIDBytes)
+	if err != nil {
+		userConn.HandleError("Internal error")
+		walletExt.Logger().Error(fmt.Errorf("error getting accounts for user (%s), %w", userID, err).Error())
+		return
+	}
+
+	found := false
+	for _, account := range accounts {
+		if bytes.Equal(account.AccountAddress, addressBytes) {
+			found = true
+		}
+	}
+
+	// create and write the response
+	res := struct {
+		Status bool `json:"status"`
+	}{Status: found}
+
+	msg, err := json.Marshal(res)
+	if err != nil {
+		userConn.HandleError("Internal error")
+		walletExt.Logger().Error(fmt.Errorf("error marshalling: %w", err).Error())
+		return
+	}
+
+	err = userConn.WriteResponse(msg)
+	if err != nil {
+		return
+	}
+}
+
+func revokeRequestHandler(walletExt *walletextension.WalletExtension, userConn userconn.UserConn) {
+	// read the request
+	_, err := userConn.ReadRequest()
+	if err != nil {
+		walletExt.Logger().Error(fmt.Errorf("error reading request: %w", err).Error())
+		return
+	}
+
+	userID, err := getQueryParameter(userConn.ReadRequestParams(), "u")
+	if err != nil {
+		userConn.HandleError("user ('u') not found in query parameters")
+		walletExt.Logger().Error(fmt.Errorf("user not found in the query params: %w", err).Error())
+		return
+	}
+	userIDBytes, err := hex.DecodeString(userID[2:]) // remove 0x prefix from userID
+	if err != nil {
+		userConn.HandleError("Internal error")
+		walletExt.Logger().Error(fmt.Errorf("error decoding string (%s), %w", userID, err).Error())
+		return
+	}
+
+	err = walletExt.Storage.DeleteUser(userIDBytes)
+	if err != nil {
+		userConn.HandleError("Internal error")
+		walletExt.Logger().Error(fmt.Errorf("error deleting user (%s), %w", userID, err).Error())
+		return
+	}
+}
```

Comment:
Should a success message be returned? Or does that happen by default ?

Reply:
success message added